lib/ecl/well_conn.cpp

```diff
@@ -157,73 +154,85 @@ well_conn_type * well_conn_alloc_from_kw( const ecl_kw_type * icon_kw ,
-        int geometric_nz = header->nz / 2;
-        if (k >= geometric_nz) {
-          k -= geometric_nz;
-          matrix_connection = false;
-        }
-      }
-    }
+  /*
+    Out in the wild we have encountered files where some of integer values used
+    to indicate direction have had invalid values. In this case we just return
+    NULL and forget about the connection - it seems to work.
+  */
+  int int_direction = ecl_kw_iget_int( icon_kw , icon_offset + ICON_DIRECTION_INDEX );
+  if ((int_direction < 0) || (int_direction > ICON_FRACY)) {
+    fprintf(stderr,"Invalid direction value:%d encountered for well - connection ignored\n",int_direction);
+    return NULL;
+  }
+
+
+  int i       = ecl_kw_iget_int( icon_kw , icon_offset + ICON_I_INDEX ) - 1;
+  int j       = ecl_kw_iget_int( icon_kw , icon_offset + ICON_J_INDEX ) - 1;
+  int k       = ecl_kw_iget_int( icon_kw , icon_offset + ICON_K_INDEX ) - 1;
+  double connection_factor = -1;
+  bool matrix_connection = true;
+  bool open;
+  well_conn_dir_enum dir = well_conn_fracX;
+
+  /* Set the status */
+  {
+    int int_status = ecl_kw_iget_int( icon_kw , icon_offset + ICON_STATUS_INDEX );
+    if (int_status > 0)
+      open = true;
+    else
+      open = false;
+  }
 
 
-    /* Set the direction flag */
-    {
-      int int_direction = ecl_kw_iget_int( icon_kw , icon_offset + ICON_DIRECTION_INDEX );
-      if (int_direction == ICON_DEFAULT_DIR_VALUE)
-        int_direction = ICON_DEFAULT_DIR_TARGET;
-
-      switch (int_direction) {
-      case(ICON_DIRX):
-        dir = well_conn_dirX;
-        break;
-      case(ICON_DIRY):
-        dir = well_conn_dirY;
-        break;
-      case(ICON_DIRZ):
-        dir = well_conn_dirZ;
-        break;
-      case(ICON_FRACX):
-        dir = well_conn_fracX;
-        break;
-      case(ICON_FRACY):
-        dir = well_conn_fracY;
-        break;
-      default:
-        util_abort("%s: icon direction value:%d not recognized\n",__func__ , int_direction);
+  /* Set the K value and fracture flag. */
+  {
+    if (header->dualp) {
+      int geometric_nz = header->nz / 2;
+      if (k >= geometric_nz) {
+        k -= geometric_nz;
+        matrix_connection = false;
       }
     }
+  }
 
-    if (scon_kw) {
-      const int scon_offset = header->nsconz * ( header->ncwmax * well_nr + conn_nr );
-      connection_factor = ecl_kw_iget_as_double(scon_kw , scon_offset + SCON_CF_INDEX);
-    }
 
+  /* Set the direction flag */
+  if (int_direction == ICON_DEFAULT_DIR_VALUE)
+    int_direction = ICON_DEFAULT_DIR_TARGET;
+
+  switch (int_direction) {
+  case(ICON_DIRX):
+    dir = well_conn_dirX;
+    break;
+  case(ICON_DIRY):
+    dir = well_conn_dirY;
+    break;
+  case(ICON_DIRZ):
+    dir = well_conn_dirZ;
+    break;
+  case(ICON_FRACX):
+    dir = well_conn_fracX;
+    break;
+  case(ICON_FRACY):
+    dir = well_conn_fracY;
+    break;
+  default:
+    util_abort("%s: icon direction value:%d not recognized\n",__func__ , int_direction);
+  }
```

Comment:
You can remove `default` now if we have the outside range covered above?

Reply:
Done.